extensions/rke1/nodetemplates/azure_config.go

```diff
@@ -19,6 +19,7 @@ type AzureNodeTemplateConfig struct {
 	ManagedDisks      bool     `json:"managedDisks" yaml:"managedDisks"`
 	NoPublicIP        bool     `json:"noPublicIp" yaml:"noPublicIp"`
 	OpenPort          []string `json:"openPort" yaml:"openPort"`
+	NSG               string   `json:"nsg" yaml:"nsg"`
```

Comment:
nsg is network security group. It's a network filter used for virtual networks. I would be iffy on adding a default value, I remember months ago that there were several complications Ugur and I saw specific to nsg.

I think just adding support for it is enough without a default value.

Reply:
yeah we don't default any values at this stage of the config so it would be best to leave it blank.

Reply:
got it thank you. I only ask because qa would have to fill in these values while creating RKE1 clusters using jenkins and if a default is provided we wouldnt have to worry about it. 

can you update these configs for this job as well https://github.com/rancherlabs/jenkins-job-builder/blob/master/qa-release-automation.yml#L560 ?

Reply:
that file isn't up to date with the last update for rke1, so I don't even think that is running at this time.

Reply:
Discussed offline and this is an optional parameter. not a breaking change and need not be updated in any of the configs.